src/System.Diagnostics.Process/tests/System.Diagnostics.Process.Tests/Given_ProcessStartInfoType.cs

```diff
@@ -0,0 +1,223 @@
+            Assert.Equal(2, index);
+
+            //Contains
+            Assert.True(Environment2.Contains(new System.Collections.Generic.KeyValuePair<string, string>("NewKey", "NewValue")));
+            if (global::Interop.IsWindows)
+            {
+                Assert.True(Environment2.Contains(new System.Collections.Generic.KeyValuePair<string, string>("nEwKeY", "NewValue")));
+            }
+            Assert.False(Environment2.Contains(new System.Collections.Generic.KeyValuePair<string, string>("NewKey99", "NewValue99")));
+
+            //Exception not thrown with invalid key
+            Assert.Throws<ArgumentNullException>(() =>
+            {
+                Environment2.Contains(new System.Collections.Generic.KeyValuePair<string, string>(null, "NewValue99"));
+            }
+            );
+
+            Environment2.Add(new System.Collections.Generic.KeyValuePair<string, string>("NewKey98", "NewValue98"));
+
+            //Indexed
+            string newIndexItem = Environment2["NewKey98"];
+            Assert.Equal("NewValue98", newIndexItem);
+
+            //TryGetValue
+            string stringout = null;
+            bool retval = false;
+            retval = Environment2.TryGetValue("NewKey", out stringout);
+            Assert.True(retval);
+            Assert.Equal("NewValue", stringout);
+            if (global::Interop.IsWindows)
+            {
+                retval = Environment2.TryGetValue("NeWkEy", out stringout);
+                Assert.True(retval);
+                Assert.Equal("NewValue", stringout);
+            }
+
+            stringout = null;
+            retval = false;
+            retval = Environment2.TryGetValue("NewKey99", out stringout);
+            Assert.Equal(null, stringout);
+            Assert.False(retval);
+
+            //Exception not thrown with invalid key
+            Assert.Throws<ArgumentNullException>(() =>
+            {
+                string stringout1 = null;
+                bool retval1 = false;
+                retval1 = Environment2.TryGetValue(null, out stringout1);
+            }
+            );
+
+            //Exception not thrown with invalid key
+            Assert.Throws<ArgumentNullException>(() =>
+            {
+                Environment2.Add(null, "NewValue2");
+            }
+            );
+
+            //Invalid Key to add
+            Assert.Throws<ArgumentException>(() =>
+            {
+                Environment2.Add("NewKey2", "NewValue2");
+            }
+            );
+            //Remove Item
+            Environment2.Remove("NewKey98");
+            Environment2.Remove("NewKey98");   //2nd occurrence should not assert
+
+            //Exception not thrown with null key
+            Assert.Throws<ArgumentNullException>(() => { Environment2.Remove(null); });
+
+            //"Exception not thrown with null key"
+            Assert.Throws<System.Collections.Generic.KeyNotFoundException>(() => { string a1 = Environment2["1bB"]; });
```

Comment:
It's not necessary to store it into a string.  We should also just import the namespace, leaving this at:
```C#
Assert.Throws<KeyNotFoundException>(() => Environment2["1bB"]);
```